internal/servers/repository_worker.go

```diff
@@ -202,18 +227,35 @@ func (r *Repository) UpsertWorkerStatus(ctx context.Context, worker *Worker, opt
 	worker = NewWorkerForStatus(worker.GetScopeId(),
 		WithName(worker.GetWorkerReportedName()),
 		WithAddress(worker.GetWorkerReportedAddress()),
-		WithWorkerTags(worker.configTags...))
+		WithWorkerTags(worker.configTags...),
+		WithKeyId(worker.WorkerReportedKeyId))
 
 	opts := getOpts(opt...)
 
 	var err error
 	switch {
 	case opts.withPublicId != "":
 		worker.PublicId = opts.withPublicId
+	case worker.WorkerReportedKeyId != "":
+		workerId, err := r.LookupWorkerIdByWorkerReportedKeyId(ctx, worker.WorkerReportedKeyId)
```

Comment:
is this path tested?

Reply:
TestStatusWorkerWithKeyId covers this; it doesn't show up when I run tests with coverage for some reason, but I've run through debug and it does hit this case.